src/ast/desugar.rs

```diff
@@ -423,15 +109,14 @@ fn add_semi_naive_rule(desugar: &mut Desugar, rule: Rule) -> Option<Rule> {
 /// type_info field is used for that.
```

Comment:
This comment is not true- I don't think type_info actually has to live in desugar anymore.
It also doesn't keep track of global variables any more.

Reply:
Great catch. I have moved `type_info` out of `Desugar`.